packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -217,6 +222,31 @@ export const CalendarMonth = ({
   const changeMonth = (newMonth: number, newYear?: number) =>
     new Date(newYear ?? focusedDate.getFullYear(), newMonth, 1);
 
+  const MIN_YEAR = 1900;
+  const MAX_YEAR = 2100;
+
+  const handleYearInputChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const yearStr = event.currentTarget.value;
+
+    if (/^\d{0,4}$/.test(yearStr)) {
```

Comment:
This is more of a nit, but just to reduce some of the nested if blocks here we could instead return early if `!/^\d{0,4}$/.test(yearStr)`.